tracer/src/Datadog.Trace/Debugger/Upload/SymbolUploadApi.cs

```diff
@@ -72,11 +82,26 @@ public override async Task<bool> SendBatchAsync(ArraySegment<byte> symbols)
             var retries = 0;
             var sleepDuration = StartingSleepDuration;
 
-            var items = new MultipartFormItem[]
+            MultipartFormItem symbolsItem;
+
+            if (_enableCompression)
+            {
+                using var memoryStream = new MemoryStream();
+#if NETFRAMEWORK
+                using var gzipStream = new Vendors.ICSharpCode.SharpZipLib.GZip.GZipOutputStream(memoryStream);
+                await gzipStream.WriteAsync(symbols.Array, 0, symbols.Array.Length).ConfigureAwait(false);
+#else
+                using var gzipStream = new GZipStream(memoryStream, CompressionMode.Compress);
+                await gzipStream.WriteAsync(symbols.Array, 0, symbols.Array.Length).ConfigureAwait(false);
+#endif
+                symbolsItem = new MultipartFormItem("file", "gzip", "file.gz", new ArraySegment<byte>(memoryStream.ToArray()));
```

Comment:
This feels like a very expensive (from an allocation PoV) way of doing the sending? 🤔 e.g. 
- We serialize everything to strings in-memory (there's basically no chance the string builder cache is going to save you from the allocations I expect, it's just going to be too big?)
- and _then_ we serialize it to an in-memory array (which we are going to resize-repeatedly as it grows unless you initialize `MemoryStream` with an array).
- and _then_ we copy that to the output stream.

A better approach _might_ be to use the `PostAsync` method that gives you a callback, so you can serialize _directly_ to the output stream.

Anyway, that's all optimisations that we could think about later, just triggered me a little bit 😄